### DISCUSSION TITLE
v2 branch has been created

### DISCUSSION THREAD

**Discussion by lestrrat** [lestrrat-go/jwx#592] [General]
No releases yet, but I believe I have amassed enough changes to warrant a v2 branch, so here it is

https://github.com/lestrrat-go/jwx/tree/v2

Feel free to take a whirl. 

ATM I don't have more (major) API changes planned, but I still need to remove github.com/pkg/errors, fix documentation, and in general give it a good review for problems

Enjoy

**Comment by lestrrat** [comment]
I have finished working on the major changes that I was planning on doing for v2
Please read the full change list here https://github.com/lestrrat-go/jwx/blob/v2/Changes.v2

I have yet to make a release, and I don't plan on doing so for a while, but NOW is your chance to propose changes!!

**Comment by cloudkucooland** [comment]
I'm working on migrating my code to v2; I can't find how to use the new jws.AutoVerify to set a jku in a token I'm minting.

Previously I'd do
    jwts, err := jwt.NewBuilder().....
    hdrs := jws.NewHeaders()
    _ = hdrs.Set(jws.JWKSetURLKey, "https://....")
    signed, err := jwt.Sign(jwts, jwa.RS256, key, jwt.WithHeaders(hdrs))

I see how to use jws.AutoVerify to verify a token that has a JKU set; I'm trying to do the opposite, generate a JWT with a JKU set in the headers. Any pointers?

**Reply by lestrrat** [reply]
I was working on that exact doc!
still need to tweak a few knobs (and I'm w/ the family now) will let you know when I'm done

**Reply by cloudkucooland** [reply]
Thanks! No rush. Other than the (probably unused by clients; intentionally not used internally) JKU being missing, my stuff works properly.

I noticed some of the docs use jws.WithInferAlgorithm when it should be jws.WithInferAlgorithmFromKey but that was easy to find.

Thanks once again for all your work on this.

**Reply by lestrrat** [reply]
PTAL if this makes sense https://github.com/lestrrat-go/jwx/blob/v2/docs/01-jwt.md#parse-and-verify-a-jwt-using-key-specified-in-jku

**Reply by cloudkucooland** [reply]
Reading the example makes perfect sense. I'll implement this in my code in a few hours to test it.

**Reply by cloudkucooland** [reply]
Builds clean. Initial tests look good.

**Reply by lestrrat** [reply]
Good. I'm switching to include actual example test code instead of pseudocode fragments in the docs so that the APIs are always in sync with the main code, hopefully it would be easier for both sides to find these things
